.github/workflows/publishDocumentation.yml

```
@@ -0,0 +1,30 @@
+name: PublishDocumentation
+on:
+  push:
+
+jobs:
+  generate-and-build-documentation:
+    runs-on: ubuntu-16.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: GenerateDocumentation
+        run: ./gradlew dokkaHtml
+
+      - name: PublishDocumentation
+        run: |
+          git config user.email "${{ secrets.CONX_GH_PAGES_USER_EMAIL }}"
+          git config user.name "${{ secrets.CONX_GH_PAGES_USER_NAME }}"
+          git fetch
+          git checkout gh-pages
+          mv core/build/dokka/html/core/ temp_core/
+          rm -rf core/
+          rm -rf com.juul.kable/
+          mv -v temp_core/* .
```

This seems to be doing unnecessary shuffling, suggest streamlining:
```suggestion
          rsync -av --delete core/build/dokka/html/core/* .
```

---

Oh, just in case the CI bot does not have `rsync` by default, alternate version to drop the extra dancing:
```
rm -rf com.juul.kable
mv -v core/build/dokka/html/core/* .
```

---

Oh nice. Yeah, I was using rsync originally. Didn't know it had a delete option; that's super nice.

---

`rsync` definitely works, but move is preferable as it's a lot more efficient.
If `mv` is too much of a struggle to get working, then `rsync` will suffice. Fortunately it is probably a pretty small amount of data anyways.

---

In this case `mv` is not particularly suitable as it requires replacing an existing non-empty directory. So it's a choice of a two-step `rm` then `mv` or use a tool that can destructively replace such as `rsync` or `cpio`.

---

@davidtaylor-juul `rsync` **copies** all the data that needs moving (which is unnecessary I/O), right?

Ultimately, small amount of data, so the I/O is fine; even if `mv` requires a little dance, IMHO it's a better choice. But overall, a nitpick, fine with either approach.

---

Yes `mv` just updates the i-nodes in the directory structure. The issue is that you cannot `mv` a folder to replace an existing folder (unless it is empty). So the existing folder(s) have to be removed first for it to work.

`rsync` makes a copy of the data but also can intelligently take care of only copying data that has changed and cleaning out files that do not appear in the source. As for which is more efficient it would entirely depend on the nature of the data set and the characteristics of the implementation of these tools on the target OS. I would not like to wager which will be faster as it could go either way.

---

Well, it all ended up being a bit of a moot point anyway cause I found a plugin that'll take care of all this shuffling for us anyway. For what it's worth, the plugin appears to be using rsync. ^_^